### PR TITLE
Travis integration.

### PR DESCRIPTION
**Pull request by mookerji**
/cc @fnoble @benjamin0 @kovach

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+sudo: required
+
+before_install:
+  - sudo add-apt-repository --yes ppa:kalakris/cmake
```
Comment (fnoble) [inline .travis.yml]: Does the Travis built in version of cmake not work?

Comment (fnoble) [inline .travis.yml]: http://docs.travis-ci.com/user/languages/c/ lists cmake as one of the built-in tools

Comment (fnoble) [inline .travis.yml]: nm, found the answers here:

https://gist.github.com/winterz/10cc2741d466cbd8ff21
https://github.com/travis-ci/travis-ci/issues/2030

might be good to add a comment linking these open travis issues

**Comment by fnoble** [comment]
otherwise, lgtm

**Comment by scottswift** [comment]
retest this please

**Comment by scottswift** [comment]
should add Check library so unit tests are run

**Pull request merged**